In [1]:
from google.colab import drive
drive.mount('/content/drive/')

Mounted at /content/drive/


In [2]:
%cd /content/drive/MyDrive/Colab Notebooks/project/UNET
%ls

/content/drive/MyDrive/Colab Notebooks/project/UNET
 before_TF_run3_checkpoint.pt   GPU_run_checkpoint_1.pth.tar   TF_run4_checkpoint.pt
 before_TF_run4_checkpoint.pt  'Kopi af golf_utils.py'         TF_run5_checkpoint.pt
 before_TF_run5_checkpoint.pt   metrics.csv                    TF_run6_checkpoint.pt
 before_TF_run6_checkpoint.pt   __pycache__/                   thickBowl_Run_6_GPU.ipynb
 checkpoint.pt                  run2_checkpoint.pt             thickPlate_Run_4_GPU.ipynb
 datasetHandler.py              Run_2_GPU.ipynb                thinBowl_Run_5_GPU.ipynb
 golfHandler.py                 run3_checkpoint.pt             thinPlate_Run_3_GPU.ipynb
 golf_predict_gpu.py            run4_checkpoint.pt             UNET_model_OG.ipynb
 golf_predict.py                run5_checkpoint.pt             unetSimple.py
 golf_train.py                  run6_checkpoint.pt             utils_temp.txt
 golf_utils.py                  TF_run3_checkpoint.pt


In [ ]:
!python golf_predict_gpu.py

Number of images: 9196
Number of masks: 9196
Number of images: 1955
Number of masks: 1955
Number of images: 1993
Number of masks: 1993
=> Loading checkpoint
Model loadet
Let's predict
Starting prediction...
Evaluating: 100% 1993/1993 [06:22<00:00,  5.21it/s]
Average Dice score: 0.9621766066249797
Average IoU score: 0.9298707861266354
Average Precision: 0.9764495680080911
Average Recall: 0.9504574458158427
Average F1 score: 0.9621766016301624
Average Accuracy: 0.9926790642247871
Average MAE: 0.00732093560462654
Average MSE: 0.00732093560462654
Average Dice Loss: 0.037823393425029904
Average BCE Loss: 0.03031782306239151
Average FPS: 40.80065761739893
Average Prediction Time per Image: 0.004621323424251727
Total Prediction Time for Epoch: 9.210297584533691
Test Dice Score: 0.9622
Test IoU Score: 0.9299
Test Precision: 0.9764
Test Recall: 0.9505
Test F1 Score: 0.9622
Test Accuracy: 0.9927
Test MAE: 0.0073
Test MSE: 0.0073
Test Dice Loss: 0.0378
Test BCE Loss: 0.0303
Test FPS: 40.80
Test A

# Second prediction

In [ ]:
!python golf_predict_gpu.py

Number of images: 9196
Number of masks: 9196
Number of images: 1955
Number of masks: 1955
Number of images: 1993
Number of masks: 1993
=> Loading checkpoint
Model loaded
Let's predict
Starting prediction...
Evaluating: 100% 1993/1993 [01:04<00:00, 30.71it/s]
Average Dice score: 0.9621766066249797
Average IoU score: 0.9298707861266354
Average Precision: 0.9764495680080911
Average Recall: 0.9504574458158427
Average F1 score: 0.9621766016301624
Average Accuracy: 0.9926790642247871
Average MAE: 0.00732093560462654
Average MSE: 0.00732093560462654
Average Dice Loss: 0.037823393425029904
Average BCE Loss: 0.03031782306239151
Average FPS: 59.43699717579891
Average Prediction Time per Image: 0.005431585436302755
Total Prediction Time for Epoch: 10.825149774551392
Test Dice Score: 0.9622
Test IoU Score: 0.9299
Test Precision: 0.9764
Test Recall: 0.9505
Test F1 Score: 0.9622
Test Accuracy: 0.9927
Test MAE: 0.0073
Test MSE: 0.0073
Test Dice Loss: 0.0378
Test BCE Loss: 0.0303
Test FPS: 59.44
Test 

# FPS fix

In [ ]:
!python golf_predict_gpu.py

Number of images: 9196
Number of masks: 9196
Number of images: 1955
Number of masks: 1955
Number of images: 1993
Number of masks: 1993
=> Loading checkpoint
Model loaded
Let's predict
Starting prediction...
Evaluating: 100% 1993/1993 [01:29<00:00, 22.15it/s]
Average Dice score: 0.9621766066249797
Average IoU score: 0.9298707861266354
Average Precision: 0.9764495680080911
Average Recall: 0.9504574458158427
Average F1 score: 0.9621766016301624
Average Accuracy: 0.9926790642247871
Average MAE: 0.00732093560462654
Average MSE: 0.00732093560462654
Average Dice Loss: 0.037823393425029904
Average BCE Loss: 0.03031782306239151
Average FPS: 166.8857046698143
Average Prediction Time per Image: 0.005992124981456705
Total Prediction Time for Epoch: 11.942305088043213
Test Dice Score: 0.9622
Test IoU Score: 0.9299
Test Precision: 0.9764
Test Recall: 0.9505
Test F1 Score: 0.9622
Test Accuracy: 0.9927
Test MAE: 0.0073
Test MSE: 0.0073
Test Dice Loss: 0.0378
Test BCE Loss: 0.0303
Test FPS: 166.89
Test

# TRACKMAN FPS FIX

In [5]:
!python golf_predict_gpu.py

Number of images: 9196
Number of masks: 9196
Number of images: 1955
Number of masks: 1955
Number of images: 112
Number of masks: 112
=> Loading checkpoint
Model loaded
Let's predict
Starting prediction...
Evaluating: 100% 112/112 [00:08<00:00, 13.93it/s]
Average Dice score: 0.7608393229568741
Average IoU score: 0.6274136051319855
Average Precision: 0.666056932951044
Average Recall: 0.9215552053386383
Average F1 score: 0.7608393181833711
Average Accuracy: 0.981678378527337
Average MAE: 0.018321621381411596
Average MSE: 0.018321621381411596
Average Dice Loss: 0.23916067715202058
Average BCE Loss: 0.05825077637564391
Average FPS: 130.09310744376577
Average Prediction Time per Image: 0.007686802319117955
Total Prediction Time for Epoch: 0.8609218597412109
Test Dice Score: 0.7608
Test IoU Score: 0.6274
Test Precision: 0.6661
Test Recall: 0.9216
Test F1 Score: 0.7608
Test Accuracy: 0.9817
Test MAE: 0.0183
Test MSE: 0.0183
Test Dice Loss: 0.2392
Test BCE Loss: 0.0583
Test FPS: 130.09
Test Ave

# Testing on 112 images from other dataset of golfers

270x480

In [ ]:
!python golf_predict_gpu.py

Number of images: 9196
Number of masks: 9196
Number of images: 1955
Number of masks: 1955
Number of images: 112
Number of masks: 112
=> Loading checkpoint
Model loaded
Let's predict
Starting prediction...
Evaluating: 100% 112/112 [00:15<00:00,  7.35it/s]
Average Dice score: 0.7608435250544139
Average IoU score: 0.6274177029843122
Average Precision: 0.6660526174150834
Average Recall: 0.9215768318411784
Average F1 score: 0.7608435202809394
Average Accuracy: 0.9816788607804235
Average MAE: 0.018321139141335152
Average MSE: 0.018321139141335152
Average Dice Loss: 0.2391564787498542
Average BCE Loss: 0.05825170520360449
Average FPS: 14.72121854656115
Average Prediction Time per Image: 0.009992737855230058
Total Prediction Time for Epoch: 1.1191866397857666
Test Dice Score: 0.7608
Test IoU Score: 0.6274
Test Precision: 0.6661
Test Recall: 0.9216
Test F1 Score: 0.7608
Test Accuracy: 0.9817
Test MAE: 0.0183
Test MSE: 0.0183
Test Dice Loss: 0.2392
Test BCE Loss: 0.0583
Test FPS: 14.72
Test Aver

# Testing on 112 images from other dataset of golfers

160x160 test images 4

In [ ]:
!python golf_predict_gpu.py

Number of images: 9196
Number of masks: 9196
Number of images: 1955
Number of masks: 1955
Number of images: 112
Number of masks: 112
=> Loading checkpoint
Model loaded
Let's predict
Starting prediction...
Evaluating: 100% 112/112 [00:09<00:00, 12.02it/s]
Average Dice score: 0.7513867638926863
Average IoU score: 0.6683942699395444
Average Precision: 0.7055748733077918
Average Recall: 0.8186829945284997
Average F1 score: 0.751386759597298
Average Accuracy: 0.9856762695312495
Average MAE: 0.014323730233757357
Average MSE: 0.014323730233757357
Average Dice Loss: 0.16825609228440694
Average BCE Loss: 0.046599386226651926
Average FPS: 54.34957839808047
Average Prediction Time per Image: 0.007362714835575649
Total Prediction Time for Epoch: 0.8246240615844727
Test Dice Score: 0.7514
Test IoU Score: 0.6684
Test Precision: 0.7056
Test Recall: 0.8187
Test F1 Score: 0.7514
Test Accuracy: 0.9857
Test MAE: 0.0143
Test MSE: 0.0143
Test Dice Loss: 0.1683
Test BCE Loss: 0.0466
Test FPS: 54.35
Test Ave

# Transfer Learning

In [ ]:
!python golf_train.py

Number of images: 70
Number of masks: 70
/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:558: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.
  warnings.warn(_create_warning_msg(
Number of images: 30
Number of masks: 30
Number of images: 12
Number of masks: 12
Print loading model...
=> Loading checkpoint
Model loaded
Let's train
Epoch 1/100
100% 70/70 [00:34<00:00,  2.04it/s, loss=0.0523]
Evaluating: 100% 30/30 [00:25<00:00,  1.17it/s]
Average Dice score: 0.8267035971455179
Average IoU score: 0.714415876070919
Average Precision: 0.8196579841154216
Average Recall: 0.8498540113657854
Average F1 score: 0.8267035921958819
Average Accuracy: 0.9879158950617284


In [ ]:
!python golf_predict_gpu.py

Number of images: 70
Number of masks: 70
Number of images: 30
Number of masks: 30
Number of images: 12
Number of masks: 12
=> Loading checkpoint
Model loaded
Let's predict
Starting prediction...
Evaluating: 100% 12/12 [00:10<00:00,  1.14it/s]
Average Dice score: 0.8976447333266765
Average IoU score: 0.8287082515686263
Average Precision: 0.9425085108947603
Average Recall: 0.8752714457605605
Average F1 score: 0.8976447284013492
Average Accuracy: 0.9953253600823045
Average MAE: 0.00467463985842187
Average MSE: 0.00467463985842187
Average Dice Loss: 0.10235527157783508
Average BCE Loss: 0.017406352912075818
Average FPS: 8.951752293044416
Average Prediction Time per Image: 0.03639596700668335
Total Prediction Time for Epoch: 0.4367516040802002
Test Dice Score: 0.8976
Test IoU Score: 0.8287
Test Precision: 0.9425
Test Recall: 0.8753
Test F1 Score: 0.8976
Test Accuracy: 0.9953
Test MAE: 0.0047
Test MSE: 0.0047
Test Dice Loss: 0.1024
Test BCE Loss: 0.0174
Test FPS: 8.95
Test Average Prediction